notebook/notebookapp.py

```diff
@@ -1590,6 +1631,13 @@ def init_configurables(self):
         )
 
     def init_logging(self):
+        if self.log_json:
+            self.log.debug('initializing json logging')
+            # Note that we don't guard against ImportError here because if the
+            # package is missing then _validate_log_json should have set
+            # log_json=False.
+            import json_logging
+            json_logging.init_non_web(enable_json=True)
```

Comment:
Let's move this into the validation block.  This will also produce json-formatted logging for the initial set of logged statements, like these...
```
{"written_at": "2020-10-09T21:10:49.519Z", "written_ts": 1602277849519721000, "msg": "Searching ['/opt/anaconda3/envs/notebook-dev/bin', '/Users/kbates/.jupyter', '/opt/anaconda3/envs/nb5386/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 179}
{"written_at": "2020-10-09T21:10:49.520Z", "written_ts": 1602277849520456000, "msg": "Looking for jupyter_config in /etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.520Z", "written_ts": 1602277849520942000, "msg": "Looking for jupyter_config in /usr/local/etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.521Z", "written_ts": 1602277849521220000, "msg": "Looking for jupyter_config in /opt/anaconda3/envs/nb5386/etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.521Z", "written_ts": 1602277849521473000, "msg": "Looking for jupyter_config in /Users/kbates/.jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.521Z", "written_ts": 1602277849521719000, "msg": "Looking for jupyter_config in /opt/anaconda3/envs/notebook-dev/bin", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.523Z", "written_ts": 1602277849523903000, "msg": "Looking for jupyter_notebook_config in /etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.524Z", "written_ts": 1602277849524180000, "msg": "Looking for jupyter_notebook_config in /usr/local/etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.524Z", "written_ts": 1602277849524445000, "msg": "Looking for jupyter_notebook_config in /opt/anaconda3/envs/nb5386/etc/jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.524Z", "written_ts": 1602277849524692000, "msg": "Looking for jupyter_notebook_config in /Users/kbates/.jupyter", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.526Z", "written_ts": 1602277849526005000, "msg": "Loaded config file: /Users/kbates/.jupyter/jupyter_notebook_config.py", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 578}
{"written_at": "2020-10-09T21:10:49.526Z", "written_ts": 1602277849526326000, "msg": "Looking for jupyter_notebook_config in /opt/anaconda3/envs/notebook-dev/bin", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "application", "line_no": 556}
{"written_at": "2020-10-09T21:10:49.529Z", "written_ts": 1602277849529904000, "msg": "initializing json logging", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "notebookapp", "line_no": 1636}
{"written_at": "2020-10-09T21:10:49.541Z", "written_ts": 1602277849541371000, "msg": "Paths used for configuration of jupyter_notebook_config: __/etc/jupyter/jupyter_notebook_config.json", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "config_manager", "line_no": 97}
{"written_at": "2020-10-09T21:10:49.542Z", "written_ts": 1602277849542273000, "msg": "Paths used for configuration of jupyter_notebook_config: __/usr/local/etc/jupyter/jupyter_notebook_config.json", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "config_manager", "line_no": 97}
{"written_at": "2020-10-09T21:10:49.543Z", "written_ts": 1602277849543104000, "msg": "Paths used for configuration of jupyter_notebook_config: __/opt/anaconda3/envs/nb5386/etc/jupyter/jupyter_notebook_config.json", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "config_manager", "line_no": 97}
{"written_at": "2020-10-09T21:10:49.543Z", "written_ts": 1602277849543888000, "msg": "Paths used for configuration of jupyter_notebook_config: __/Users/kbates/.jupyter/jupyter_notebook_config.json", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "DEBUG", "module": "config_manager", "line_no": 97}
{"written_at": "2020-10-09T21:10:49.587Z", "written_ts": 1602277849587112000, "msg": "It looks like you're running the notebook from source._    If you're working on the Javascript of the notebook, try running__    npm run build:watch__    in another terminal window to have the system incrementally_    watch and build the notebook's JavaScript for you, as you make changes.", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 232}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647173000, "msg": "Serving notebooks from local directory: /Users/kbates/notebooks/alice", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647428000, "msg": "Jupyter Notebook 7.0.0.dev0 is running at:", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647604000, "msg": "http://Kevins-MacBook-Pro.local:8888/?token=e3275dfe8643b393967cdd1d181fde6ef60e2f1a100e98d6", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647719000, "msg": " or http://127.0.0.1:8888/?token=e3275dfe8643b393967cdd1d181fde6ef60e2f1a100e98d6", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647819000, "msg": "Kernels will be managed by the Gateway server running at:", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.647Z", "written_ts": 1602277849647914000, "msg": "http://yarn-eg-node-1.fyre.ibm.com:8888", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2258}
{"written_at": "2020-10-09T21:10:49.648Z", "written_ts": 1602277849648021000, "msg": "Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2259}
{"written_at": "2020-10-09T21:10:49.648Z", "written_ts": 1602277849648127000, "msg": "Welcome to Project Jupyter! Explore the various tools available and their corresponding documentation. If you are interested in contributing to the platform, please visit the communityresources section at https://jupyter.org/community.html.", "type": "log", "logger": "NotebookApp", "thread": "MainThread", "level": "INFO", "module": "notebookapp", "line_no": 2261}
```

Reply:
Done.